apps/src/labs/projects/ProjectManager.ts

```diff
@@ -111,14 +116,22 @@ export default class ProjectManager {
       return this.getNoopResponseAndSendSaveNoopEvent();
     }
     this.sourceToSave = source;
-    if (!this.canSave(forceSave)) {
-      if (!this.saveQueued) {
-        this.enqueueSave();
-      }
+    return this.enqueueSaveOrSave(forceSave);
+  }
+
+  async rename(name: string, forceSave = false) {
+    if (this.destroyed || !this.lastChannel) {
+      // If we have already been destroyed or the channel does not exist,
+      // don't attempt to rename.
       return this.getNoopResponseAndSendSaveNoopEvent();
-    } else {
-      this.saveHelper();
     }
+    if (!this.channelToSave) {
+      this.channelToSave = JSON.parse(
+        JSON.stringify(this.lastChannel)
+      ) as Channel;
+    }
+    this.channelToSave.name = name;
```

Comment:
Are there fields other than `name` that can be updated client-side? We could potentially narrow this and only keep track of the new name to save rather than updating and comparing the whole channelToSave object

Reply:
There are a bunch of fields that we could end up using (these are all mentioned in project.js):
`level, frozen, hidden, thumbnailUrl, migratedToS3, sharedWith, libraryName, libraryDescription, latestLibraryVersion, publishLibrary, libraryPublishedAt`, which is why I wanted to keep it generic for ease of updating in the future.